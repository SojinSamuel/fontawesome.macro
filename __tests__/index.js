const pluginTester = require('babel-plugin-tester').default

const plugin = require('babel-plugin-macros')

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  tests: [
    {
      title: 'can import macros for free version',
      snapshot: true,
      code: `
        const { far, fas, fal, fab } = require('./fixtures/fontawesome5.macro.js')
      `
    },
    {
      title: 'can use string template syntax',
      snapshot: true,
      code: `
        const { far } = require('./fixtures/fontawesome5.macro.js')
        far\`smile\`
      `
    },
    {
      title: 'can use twice',
      snapshot: true,
      code: `
        const { far } = require('./fixtures/fontawesome5.macro.js')
        far\`smile\`
        far\`smile\`
      `
    },
    {
      title: 'can use function syntax',
      snapshot: true,
      code: `
        const { far } = require('./fixtures/fontawesome5.macro.js')
        far('smile')
      `
    },
    {
      title: 'can mix both syntaxes',
      snapshot: true,
      code: `
        const { far } = require('./fixtures/fontawesome5.macro.js')
        far('smile')
        far\`smile\`
      `
    }
  ]
})
