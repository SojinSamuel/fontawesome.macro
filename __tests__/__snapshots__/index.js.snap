// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros can import macros for free version: can import macros for free version 1`] = `

const { far, fas, fal, fab } = require('./fixtures/fontawesome5.macro.js')

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros can mix both syntaxes: can mix both syntaxes 1`] = `

const { far } = require('./fixtures/fontawesome5.macro.js')
far('smile')
far\`smile\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { faSmile as _farSmile2 } from '@fortawesome/free-regular-svg-icons'
import { faSmile as _farSmile } from '@fortawesome/free-regular-svg-icons'
_farSmile
_farSmile2


`;

exports[`macros can use function syntax: can use function syntax 1`] = `

const { far } = require('./fixtures/fontawesome5.macro.js')
far('smile')

      ↓ ↓ ↓ ↓ ↓ ↓

import { faSmile as _farSmile } from '@fortawesome/free-regular-svg-icons'
_farSmile


`;

exports[`macros can use string template syntax: can use string template syntax 1`] = `

const { far } = require('./fixtures/fontawesome5.macro.js')
far\`smile\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { faSmile as _farSmile } from '@fortawesome/free-regular-svg-icons'
_farSmile


`;

exports[`macros can use twice: can use twice 1`] = `

const { far } = require('./fixtures/fontawesome5.macro.js')
far\`smile\`
far\`smile\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { faSmile as _farSmile2 } from '@fortawesome/free-regular-svg-icons'
import { faSmile as _farSmile } from '@fortawesome/free-regular-svg-icons'
_farSmile
_farSmile2


`;
